### ISSUE TITLE
How to capture incoming request?

### ISSUE THREAD

**Issue by pkellner** [graphql-dotnet/graphql-dotnet#1464]
I'm still using asp.net core 2.2 and following the example from the Pluralsight course for setting up GraphQL.  That is, my `Startup.cs` is basically the following. I want to log the incoming request to /graphQL so I can tell what is going wrong. 

Are there any entry points . I can hook into to monitor (so I can output) the incoming GraphQl request?

Here is my startup

```
            app.Use((context, next) =>
            {
                context.Response.Headers["Access-Control-Allow-Origin"] = "*";
                context.Response.Headers["Access-Control-Allow-Headers"] =
                    "Origin, X-Requested-With, Content-Type, Accept, Authorization";
                context.Response.Headers["Access-Control-Allow-Methods"] = "GET, POST, PUT, PATCH, DELETE, OPTIONS";
                context.Response.Headers["Access-Control-Allow-Credentials"] = "true";
                return next.Invoke();
            });

            app.UseWebSockets();
            app.UseGraphQLWebSockets<CarvedRockSchema>("/graphql");
```

and in my Configure Services

```
            services.AddGraphQL(o => { o.ExposeExceptions = _env.IsDevelopment(); })
                .AddGraphTypes(ServiceLifetime.Scoped).AddUserContextBuilder(httpContext => httpContext.User)
                .AddDataLoader().AddWebSockets();
```

**Comment by sungam3r** [comment]
To log an incoming request, you can use any tool convenient for you. GraphQL in this respect is not something special that prevents you from log the text/headers of the incoming request. For example you can use [this](https://exceptionnotfound.net/using-middleware-to-log-requests-and-responses-in-asp-net-core/).

**Comment by SimonCropp** [comment]
seems this one is answered and should be closed?

**Comment by sungam3r** [comment]
I give such questions `needs confirmation` label and then after about a month I close it, if there is no feedback.

**Comment by reactatscale** [comment]
Close
